ckanext/canada/activity.py

```diff
@@ -0,0 +1,77 @@
+from ckantoolkit import h
+from ckan.logic import get_action
+from ckan.common import _
+from ckan.lib.helpers import url_for
+import datetime
+
+
+def datastore_activity_create(context, act_data):
+    activity_type = act_data['activity_type']
+    count = act_data['count']
+    resource_id = act_data['resource_id']
+    user = context['user']
+    model = context['model']
+    user_id = model.User.by_name(user.decode('utf8')).id
+    if activity_type == 'deleted datastore':
+        #  get last deleted activity for this user, if within 2 seconds,
+        #   merge these activities
+        act = model.activity.user_activity_list(user_id, 1, 0)
+        now = datetime.datetime.now()
+        if act and len(act)>0 and (
+                now - act[0].timestamp).total_seconds()<2 and (
```

Comment:
use a constant defined at the top of the module like `MERGE_ACTIVITIES_WITHIN_SECONDS = 2`